lib/pages/home/home_page.dart

```diff
@@ -0,0 +1,183 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_survey/gen/assets.gen.dart';
+import 'package:flutter_survey/pages/widgets/dimmed_background.dart';
+import 'package:flutter_survey/resources/dimens.dart';
+import 'package:page_view_indicators/circle_page_indicator.dart';
+
+class HomePage extends StatefulWidget {
+  @override
+  HomePageState createState() {
+    return new HomePageState();
+  }
+}
+
+class HomePageState extends State<HomePage> {
+  // TODO
+  final _surveys = [1, 2, 3];
+  final _pageController = PageController();
+  final _currentPageNotifier = ValueNotifier<int>(0);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      body: Stack(
+        children: <Widget>[
+          _buildSurveyPageViewer(),
+          SafeArea(
+            child: Padding(
+              padding: EdgeInsets.all(Dimens.defaultMarginPadding),
+              child: _buildSurveyDetail(context),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildSurveyPageViewer() {
+    return PageView.builder(
+      itemCount: _surveys.length,
+      controller: _pageController,
+      itemBuilder: (BuildContext context, int index) {
+        return _buildSurveyPageView(context);
+      },
+      onPageChanged: (int index) {
+        _currentPageNotifier.value = index;
+      },
+    );
+  }
+
+  Widget _buildSurveyPageView(BuildContext context) {
+    return Stack(
+      children: [
+        Container(
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: Assets.images.bgHomeSample,
+              fit: BoxFit.cover,
+            ),
+          ),
+        ),
+        _buildDimmedBackground(),
+      ],
+    );
+  }
```

Comment:
@luongvo imo, we should break the main widget function to separate widgets to easily manage their components instead of creating the widget functions like this. So we can avoid `home_page` becoming a huge widget in the future.
My suggestion is to create a `SurveyPageViewer` widget, and a `SurveyDetail` widget. If they are still complex, we can break more. How do you think? 🙏

Reply:
@manh-t looks like creating adapter, item view in Android native huh? Let me try so

Reply:
@luongvo `SurveyDetail` is another widget like this implementation 😃 
![image](https://user-images.githubusercontent.com/60863885/133199749-2dd7b43b-ac9a-4ecb-952d-3baf41e52587.png)

Reply:
@manh-t I know that, no worries 👋